Cue/app/reducers/library.js

```diff
@@ -39,16 +39,16 @@ function library(state: State = initialState, action: Action): State {
     }
 
   } else if (action.type == 'DECK_EDITED') {
-    let editedDeck = action.deck
+    let change = action.change
     decks.find( (deck, i) => {
-      if (deck.uuid == editedDeck.uuid)
-        return decks[i] = { ...deck, editedDeck }
+      if (deck.uuid == change.uuid)
+        return decks[i] = { ...deck, change }
     })
     if (!localChanges.find( (deck, i) => {
-      if (deck.uuid == editedDeck.uuid)
-        return localChanges[i] = { ...deck, editedDeck }
+      if (deck.uuid == change.uuid)
+        return localChanges[i] = { ...deck, change, action: deck.action }
```

Comment:
Ah, yes. You would need to unpack them both, otherwise it would just be the same as writing `{...deck, change: change}`. I believe this only does a *shallow merge* though, so only the latest set of changes will be picked up.

You might need to manually merge the two `deck.cards` and `change.cards` arrays. If you just concatenate them both in order (i.e., `{...deck, ...change, cards: [...deck.cards, ...change.cards]}`, I think that should also work as long as the server is able to process successive edits to a single card in a single request.